storm-server/src/main/java/org/apache/storm/scheduler/blacklist/strategies/IBlacklistStrategy.java

```diff
@@ -35,7 +36,9 @@
      *       the `cluster` object.
      * @return blacklisted supervisors' id set
      */
-    Set<String> getBlacklist(List<Map<String, Set<Integer>>> badSupervisorsToleranceSlidingWindow, Cluster cluster, Topologies topologies);
+    Set<String> getBlacklist(List<Map<String, Set<Integer>>> badSupervisorsToleranceSlidingWindow,
```

Comment:
Looks like we need to modify the interface every time if we want to add another kind of failures.

Technically this is a public interface and users can provide their own implementation so backwards compatibility should be preserved as long as possible. If we can make it backwards compatible, that would be fantastic.

But I assume users don't really do that since this is actually tightly coupled and not really easy for users to provide their own implementation; even if they do, it is easy to make the change since the interface change is minimum. I feel this `IBlacklistStrategy` interface itself is not well defined and there are more refactoring to be done with this interface. So if it's hard to maintain backwards compatibility, this change is fine to me. Refactoring can be a follow-up